### PR TITLE
Implement EDTF parsing, validation, and conversion

### PR DESCRIPTION
**Pull request by bmquinn**
- Adds a top-level `EDTF` module (which is a `GenServer`) that implements `validate/1`, `parse/1`, and `humanize/1` with full integration testing and public interface documentation
- Adds `priv/edtf` node application that is called via a port from the `EDTF` module

### CODE REVIEW

File: config/config.exs [hunk 1]
```diff
@@ -55,6 +55,11 @@ with val <- System.get_env("PYRAMID_PROCESSOR") do
   unless is_nil(val), do: config(:meadow, pyramid_processor: val)
 end
 
+# Configures the pyramid TIFF processor
```
Comment (kdid) [inline config/config.exs:58]: update comment?

**Comment by kdid** [comment]
Can you wire it up to the API?
https://github.com/nulib/meadow/blob/deploy/staging/lib/meadow/data/works.ex#L499

(2nd done looks like in the issue)

File: config/config.exs (hunk 1 above)
Comment (mbklein) [reply config/config.exs]: Fixed

**Comment by kdid** [comment]
When I try something that is not a valid EDTF date through the API it's throwing an internal server error. I would expect a validation error to bubble back up through GraphQL like it does for other invalid fields.

Example:

```
mutation{
  createWork(
    accessionNumber: "wfeeddeodddofw4dd44w",
    descriptiveMetadata: {
      dateCreated: [{edtf: "abcd"}],
    },
    administrativeMetadata: {}
  ){
    id
    descriptiveMetadata{
      dateCreated{
        humanized
        edtf
      }
    }
  }
}
```


```
[error] #PID<0.1686.0> running MeadowWeb.Endpoint (connection #PID<0.1458.0>, stream id 45) terminated
Server: devbox.library.northwestern.edu:3001 (https)
Request: POST /api/graphiql
** (exit) an exception was raised:
    ** (Protocol.UndefinedError) protocol String.Chars not implemented for {:array, Meadow.Data.Types.EDTFDate} of type Tuple. This protocol is implemented for the following type(s): Phoenix.LiveComponent.CID, Postgrex.Copy, Postgrex.Query, Decimal, Float, DateTime, Time, List, Version.Requirement, Atom, Integer, Version, Date, BitString, NaiveDateTime, URI
        (elixir 1.10.4) lib/string/chars.ex:3: String.Chars.impl_for!/1
        (elixir 1.10.4) lib/string/chars.ex:22: String.Chars.to_string/1
        (meadow 0.1.0) lib/meadow_web/schema/changeset_errors.ex:11: anonymous fn/2 in MeadowWeb.Schema.ChangesetErrors.error_details/1
        (elixir 1.10.4) lib/enum.ex:2111: Enum."-reduce/3-lists^foldl/2-0-"/3
        (ecto 3.5.3) lib/ecto/changeset.ex:2888: anonymous fn/3 in Ecto.Changeset.merge_error_keys/3
        (elixir 1.10.4) lib/enum.ex:2111: Enum."-reduce/3-lists^foldl/2-0-"/3
        (ecto 3.5.3) lib/ecto/changeset.ex:2882: Ecto.Changeset.traverse_errors/2
        (ecto 3.5.3) lib/ecto/changeset.ex:2922: anonymous fn/4 in Ecto.Changeset.merge_related_keys/4
        (stdlib 3.13) maps.erl:233: :maps.fold_1/3
        (meadow 0.1.0) lib/meadow_web/resolvers/data.ex:25: MeadowWeb.Resolvers.Data.create_work/3
        (absinthe 1.5.3) lib/absinthe/resolution.ex:209: Absinthe.Resolution.call/2
        (absinthe 1.5.3) lib/absinthe/phase/document/execution/resolution.ex:230: Absinthe.Phase.Document.Execution.Resolution.reduce_resolution/1
        (absinthe 1.5.3) lib/absinthe/phase/document/execution/resolution.ex:185: Absinthe.Phase.Document.Execution.Resolution.do_resolve_field/3
        (absinthe 1.5.3) lib/absinthe/phase/document/execution/resolution.ex:170: Absinthe.Phase.Document.Execution.Resolution.do_resolve_fields/6
        (absinthe 1.5.3) lib/absinthe/phase/document/execution/resolution.ex:88: Absinthe.Phase.Document.Execution.Resolution.walk_result/5
        (absinthe 1.5.3) lib/absinthe/phase/document/execution/resolution.ex:67: Absinthe.Phase.Document.Execution.Resolution.perform_resolution/3
        (absinthe 1.5.3) lib/absinthe/phase/document/execution/resolution.ex:24: Absinthe.Phase.Document.Execution.Resolution.resolve_current/3
        (absinthe 1.5.3) lib/absinthe/pipeline.ex:368: Absinthe.Pipeline.run_phase/3
        (absinthe_plug 1.5.0) lib/absinthe/plug.ex:445: Absinthe.Plug.run_query/4
        (absinthe_plug 1.5.0) lib/absinthe/plug.ex:258: Absinthe.Plug.call/2
```

**Comment by mbklein** [comment]
Ah, I didn't realize that wasn't already being handled downstream. I'll take another pass at it.

**Pull request merged**